darts/utils/data/encoders.py

```diff
@@ -0,0 +1,570 @@
+        -------
+        Tuple[past_covariate, future_covariate]
+            The past_covariate and/or future_covariate for prediction/inference including the encodings.
+            If input {x}_covariate is None and no {x}_encoders are given, will return `None`
+            for the {x}_covariate.
+        """
+
+        return self._launch_encoder(target=target,
+                                    past_covariate=past_covariate,
+                                    future_covariate=future_covariate,
+                                    n=n,
+                                    encode_past=encode_past,
+                                    encode_future=encode_future)
+
+    def _launch_encoder(self,
+                        target: Sequence[TimeSeries],
+                        past_covariate: SupportedTimeSeries,
+                        future_covariate: SupportedTimeSeries,
+                        n: Optional[int] = None,
+                        encode_past: bool = True,
+                        encode_future: bool = True) -> Tuple[Sequence[TimeSeries], Sequence[TimeSeries]]:
+        """Launches the encode sequence for past covariate and future covariate for either training or
+        inference/prediction.
+
+        If `n` is `None` it is a prediction, otherwise it is training.
+        """
+
+        if not self.encoding_available:
+            return past_covariate, future_covariate
+
+        target = [target] if isinstance(target, TimeSeries) else target
+
+        if self.past_encoders and encode_past:
+            past_covariate = self._encode_sequence(encoders=self.past_encoders,
+                                                   target=target,
+                                                   covariate=past_covariate,
+                                                   n=n)
+
+        if self.future_encoders and encode_future:
+            future_covariate = self._encode_sequence(encoders=self.future_encoders,
+                                                     target=target,
+                                                     covariate=future_covariate,
+                                                     n=n)
+
+        return past_covariate, future_covariate
+
+    def _encode_sequence(self,
+                         encoders: Sequence[SingleEncoder],
+                         target: Sequence[TimeSeries],
+                         covariate: Optional[SupportedTimeSeries],
+                         n: Optional[int] = None) -> Sequence[TimeSeries]:
+        """Sequentially encodes the index of all input target/covariate TimeSeries
+
+        If `n` is `None` it is a prediction and method `encoder.encode_inference()` is called.
+        Otherwise, it is a training case and `encoder.encode_train()` is called.
+        """
+
+        encode_method = 'encode_train' if n is None else 'encode_inference'
+
+        encoded_sequence = []
+        if covariate is None:
+            covariate = [None] * len(target)
```

Comment:
If I understand well, these lists have size > 1 only in the "eager" case, right?

Reply:
yes :)